internal/provider/deployment.go

```diff
@@ -97,15 +97,15 @@ func getDeploymentDeployer(d *schema.ResourceData, apiClient *apiClient) (Deploy
 		NetworkName: networkName,
 		APIClient:   apiClient,
 		ncPool:      pool,
-		deployer:    deployer.NewDeployer(apiClient.identity, apiClient.twin_id, apiClient.grid_client, pool, true, solutionProvider, string(deploymentDataStr)),
+		deployer:    deployer.NewDeployer(apiClient.identity, apiClient.twin_id, apiClient.grid_client, pool, true, nil, string(deploymentDataStr)),
 	}
```

Comment:
line 81: we are logging the error when we fail to marshal the deployment data instead of returning the error

Reply:
on the NewDeployer call, why did we pass nil for the solutionProvider?